frontend/app/helpers/spree/frontend_helper.rb

```diff
@@ -368,5 +374,13 @@ def checkout_edit_link(step = 'address', order = @order)
     def products_for_filters
       Product.for_filters(current_currency, @taxon)
     end
+
+    def products_for_filters_cache_key
+      [
+        products_for_filters.maximum(:updated_at).to_i,
+        current_currency,
+        @taxon&.permalink
```

Comment:
@mad-eel we also need `I18n.locale` here - or you can use the `base_cache_key` helper method - even better